File: content/search/source/includes/partial-match-tutorial/code-snippets/csharp/CreateAutoCompleteIndex.cs
using MongoDB.Bson;
using MongoDB.Driver;
using System;
using System.Threading.Tasks;

namespace MongoSearchExamples
{
    class CreateAutoCompleteIndex
    {
        // Connection URI for your Atlas deployment
        private static readonly string connectionUri = "<connection-string>";

        static async Task Main(string[] args)
        {
            var client = new MongoClient(connectionUri);
            
            try
            {
                // Set namespace
                var database = client.GetDatabase("sample_mflix");
                var collection = database.GetCollection<BsonDocument>("movies");

                // Define your MongoDB Search index
                var indexDefinition = new BsonDocument
                {
                    { "name", "partial-match-tutorial-autocomplete" },
                    { 
                        "definition", new BsonDocument
                        {
                            {
                                "mappings", new BsonDocument
                                {
                                    { "dynamic", false },
                                    {
                                        "fields", new BsonDocument
                                        {
                                            {
                                                "title", new BsonDocument
                                                {
                                                    { "type", "autocomplete" },
                                                    { "analyzer", "lucene.standard" },
                                                    { "tokenization", "edgeGram" },
                                                    { "minGrams", 3 },
                                                    { "maxGrams", 5 },
                                                    { "foldDiacritics", false }
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                };

                // Create the search index
                var result = await collection.SearchIndexes.CreateOneAsync(indexDefinition);
                Console.WriteLine($"New index name: {result}");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex.Message}");
            }
        }
    }
}

File: content/search/source/includes/partial-match-tutorial/code-snippets/java/CreateAutoCompleteIndex.java
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class CreateAutoCompleteIndex {
    public static void main(String[] args) {
        // connect to your Atlas deployment
        String uri = "<connection-string>";

        try (MongoClient mongoClient = MongoClients.create(uri)) {
            // set namespace
            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("movies");

            // define your MongoDB Search index
            Document indexDefinition = new Document("mappings", 
                new Document("dynamic", false)
                    .append("fields", 
                        new Document("title", 
                            new Document("type", "autocomplete")
                                .append("analyzer", "lucene.standard")
                                .append("tokenization", "edgeGram")
                                .append("minGrams", 3)
                                .append("maxGrams", 5)
                                .append("foldDiacritics", false)
                        )
                    )
            );

            // run the helper method
            String result = collection.createSearchIndex("partial-match-tutorial-autocomplete", indexDefinition);
            System.out.println("New index name: " + result);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
